pkg/crypto/certloader/fswatcher/fswatcher.go

```diff
@@ -343,11 +343,13 @@ func (w *Watcher) loop() {
 				}
 			}
 		case err := <-w.watcher.Errors:
-			log.WithError(err).Debug("Received fsnotify error")
+			log.WithError(err).Warn("Received fsnotify error while watching")
```

Comment:
Unrelated to the above discussion - I believe logging this with warn-level here will lead to duplicated error messages. The error is forwarded via `w.Errors` and then logged here: 

https://github.com/cilium/cilium/blob/ed650a315f764e8dc98063222c5dcbe83616c03b/pkg/crypto/certloader/watcher.go#L198-L199

Reply:
So if we're forwarding the error, I would be inclined to remove this log completely.

Reply:
My thinking was that when troubleshooting a potential bug (such as e.g. the client of this package somehow ignoring the error) that the potentially duplicate messages in debug output are worth it if the alternative is no error being logged at all. 

I don't feel super strongly about it though, feel free to remove the log statement.

Reply:
Alright, let's keep it at the debug level then!

Reply:
A quick rule of thumb is if a function returns an error, don't log. Only log when the error is "consumed", i.e. `continue` in the main loop or something like that.

Doesn't work in all cases, but a good guide to start with